tokio/src/runtime/thread_pool/worker.rs

```diff
@@ -620,8 +620,7 @@ impl Core {
         // If a task is in the lifo slot, then we must unpark regardless of
         // being notified
         if self.lifo_slot.is_some() {
-            worker.shared.idle.unpark_worker_by_id(worker.index);
-            self.is_searching = true;
+            self.is_searching = !worker.shared.idle.unpark_worker_by_id(worker.index);
```

Comment:
This is the key bit. When the worker wakes, it only enters the "searching" state if it was unparked by another thread. This distinguishes unparks from other threads to signal to the worker that it should steal work vs. the I/O driver unparked the thread because events arrived.

Reply:
it might be worth having a comment in the code explaining that? not a hard blocker though.

Reply:
done